### PR TITLE
Flexible x-scale's `bandWidth` for text on ordinal + remove `config.textCellWidth`

### LINKED ISSUE #724: Revise rules for marks
- [x] __text__: 
  - [x] `x` and `y` should work with `text`.   Currently text is only used with `row` and `col`.   — Related to #822, #741
   - [x] Property for applying color to background?
   - [x] text also current has no fill applied
 - [x] remove `textCellWidth`
- [x] __bar, area__: orientation
- [x] behavior of __size__ for different marks 
- [x] ~~square with OxO~~
- [x] `bar`’s scale (see note below)

#205, #185

### PR DESCRIPTION
**Pull request by kanitw**
- Make `x`'s `bandWidth` 90 if `mark` = `text` and `x` is ordinal scale by default  (otherwise `bandWidth` = `21` for `x` and `y`)
- Do not refer to x&y's `bandWidth` when it doesn't exist.  (And add TODO refer to the future `config.scale.bandWidth` / `config.scale.textBandWidth`)
- Remove `config.textCellWidth`, which is out of place

Fix #724

### CODE REVIEW

**Comment by domoritz** [comment]
Cool. Will merge when tests pass.

File: src/compile/layout.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ function getCellHeight(model: Model): LayoutValue {
       return model.config().cell.height;
     }
   }
-  return model.fieldDef(Y).scale.bandWidth;
+  return 21 /* config.scale.bandWidth */;
```
Comment (domoritz) [inline src/compile/layout.ts]: You're going to change all these constants, I assume.

Comment (kanitw) [inline src/compile/layout.ts]: They should refer to config.scale.width in the future.   Now it doesn't make sense to add config.scale unless we want to migrate all properties there.

Comment (domoritz) [inline src/compile/layout.ts]: Wait, so what happened to `model.fieldDef(Y).scale.bandWidth`?

Comment (kanitw) [inline src/compile/layout.ts]: It was original set in combination with positionFieldDef's definition.   (Have to be the same for both x and y) 
  
But now x scale has flexible bandWidth and going forward, we should move these default values to `config.scale.*` anyway so I decide not to bother with `encoding.x|y.scale` `s default.

**Comment by kanitw** [comment]
@domoritz  test now pass.  :) 

(And yes, I amend and force push.  Let's do this going forward when make sense.  Apologies for being dumb about this for a year!)

File: src/compile/layout.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/layout.ts]: Let's move them now. Having the same constant everywhere really scares me.

Comment (kanitw) [inline src/compile/layout.ts]: If that really scares you, it should still be done in a separate PR as that involves a lot more things.  
See #1056 about facet's scale/axis.  I don't think that belongs here and it's much messier PR than here.  

If you take a look closely, it's just 21 everywhere except a few places where it should be 90.  So it is not that many numbers.  (Just two)

Comment (domoritz) [inline src/compile/layout.ts]: Oh, I'm aware that it's only 2 numbers but they are in 16 different places. Will you send a follow up PR?

Comment (kanitw) [inline src/compile/layout.ts]: I also added comment everywhere so everyone is warned not to change them!

Comment (domoritz) [inline src/compile/layout.ts]: Yep, saw that. Do you want to send a follow up PR?

Comment (kanitw) [inline src/compile/layout.ts]: No time to do it yet...
